astropy/units/quantity.py

```diff
@@ -445,7 +445,12 @@ def __new__(
         if float_default:
             dtype = None
 
-        # optimize speed for Quantity with no dtype given, copy=False
+        if not NUMPY_LT_2_0 and copy is False:
+            # strict backward compatibility
+            # see https://github.com/astropy/astropy/pull/16142
+            copy = None
```

Comment:
I think beyond this point any time one has `copy=False`, one should set `copy=COPY_IF_NEEDED` instead, so that we avoid doing this another time. Also, as is, l.458 would lead to a problem at l.466...